plugin/kubernetes/object/service.go

```diff
@@ -47,6 +47,8 @@ func ToService(obj interface{}) interface{} {
 		copy(s.Ports, svc.Spec.Ports)
 	}
 
+	svc = nil
```

Comment:
Can you help me understand this better?  ...

Per my understanding, `svc` is a local scope pointer ... nothing points to _it_, so it should get allocated on the stack.  It goes away when `ToService(obj)` pops off the stack.  The object `svc` points to is still pointed to by `obj` (which is also on the stack and will go away when the function ends).  The object pointed to would still pointed to by the caller to `ToService(obj)` ... So, I don't see how setting it to nil here makes any difference.